api/pkg/plugins/plugins.go

```diff
@@ -188,6 +191,24 @@ func (cp *PluginConfigParser) Parse(any interface{}) (res interface{}, err error
 	return conf, nil
 }
 
+func RegisterMetricsCallback(pluginName string, registerMetricFunc func(capi.ConfigCallbacks)) {
```

Comment:
Maybe we can define an interface for plugin that has metrics defined? Like:
https://github.com/mosn/htnn/blob/982b45c38b7acdcae32d2627ccd47f9a5c41453c/api/pkg/filtermanager/config.go#L246

So we don't need an extra RegisterMetricsCallback.

Reply:
addressed in the new commit